sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -415,13 +417,21 @@ case class AlterTableChangeColumnCommand(
     Seq.empty[Row]
   }
 
-  // Find the origin column from schema by column name, throw an AnalysisException if the column
-  // reference is invalid.
+  // Find the origin column from schema by column name
   private def findColumnByName(
-      schema: StructType, name: String, resolver: Resolver): StructField = {
+      schema: StructType, name: String, resolver: Resolver): Option[StructField] = {
     schema.fields.collectFirst {
       case field if resolver(field.name, name) => field
-    }.getOrElse(throw QueryCompilationErrors.cannotFindColumnError(name, schema.fieldNames))
+    }
+  }
+
+  private def verifyNonPartitionColumn(
+      table: CatalogTable, columnName: String, resolver: Resolver): Unit = {
+    findColumnByName(table.partitionSchema, columnName, resolver) match {
+      case Some(_) =>
+        throw QueryCompilationErrors.cannotAlterPartitionColumn(table.qualifiedName, columnName)
+      case None =>
+    }
```

Comment:
Could you make it shorter:
```suggestion
    if (findColumnByName(table.dataSchema, columnName, resolver).isDefined) {
      throw QueryCompilationErrors.cannotAlterPartitionColumn(table.qualifiedName, columnName)
    }
```